kyo-prelude/shared/src/test/scala/kyo/EmitTest.scala

```diff
@@ -69,162 +67,114 @@ class EmitTest extends Test:
         assert(res.eval == (Chunk(Set(1), Set(2), Set(3)), "a"))
     }
 
-    "Ack" - {
-        "apply" - {
-            "negative" in {
-                val ack = Ack(-1)
-                assert(ack == Ack.Stop)
-            }
-            "zero" in {
-                val ack = Ack(0)
-                assert(ack == Ack.Stop)
-            }
-            "positive" in {
-                Ack(1) match
-                    case Continue(n) => assert(n == 1)
-                    case Stop        => fail()
-            }
+    "runForeach" - {
+        "with pure function" in {
+            var seen = List.empty[Int]
+            def emits(i: Int): Unit < Emit[Int] =
+                if i == 5 then ()
+                else Emit.valueWith(i)(emits(i + 1))
+            end emits
+
+            Emit.runForeach { (v: Int) =>
```

Comment:
@fwbrasil  Now that the order is switched you have to explicitly annotate the parameter type for it to compile.